lib/datadog/appsec/processor.rb

```diff
@@ -22,7 +22,9 @@ def run(input, timeout = WAF::LibDDWAF::DDWAF_RUN_TIMEOUT)
 
           start_ns = Core::Utils::Time.get_time(:nanosecond)
 
-          _code, res = @context.run(input, timeout)
+          cleaned_input = remove_empty_entries(input)
```

Comment:
Looks like `remove_empty_entries` is merely `.reject { |k, v| v.nil? || v.empty? }` or did I miss something?

Also since `input` is thrown away in that case I would suggest mutating in place:

```suggestion
          input.reject! { |_, v| v.nil? || v.empty? }
```